crypto/evp_extra/pq_sig_kat_tests_dilithium3.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <openssl/evp.h>
+#include <vector>
+#include "../fipsmodule/evp/internal.h"
+#include "../dilithium/sig_dilithium.h"
+#include "../test/file_test.h"
+#include "../test/test_util.h"
+#include "../rand_extra/pq_custom_randombytes.h"
+#include "./internal.h"
+
+ static void RunTest(FileTest *t)
+{
+  std::string count, mlen, smlen;
+  std::vector<uint8_t> seed, msg, pk, sk, sm;
+
+  ASSERT_TRUE(t->GetAttribute(&count, "count"));
+  ASSERT_TRUE(t->GetBytes(&seed, "seed"));
+  ASSERT_TRUE(t->GetAttribute(&mlen, "mlen"));
+  ASSERT_TRUE(t->GetBytes(&msg, "msg"));
+  ASSERT_TRUE(t->GetBytes(&pk, "pk"));
+  ASSERT_TRUE(t->GetBytes(&sk, "sk"));
+  ASSERT_TRUE(t->GetAttribute(&smlen, "smlen"));
+  ASSERT_TRUE(t->GetBytes(&sm, "sm"));
+
+  //the KAT files generated by the dilithium team use the optional APIs that
+  //create a signature for a message m and append the message to the end of the
+  //signature. We only want to bring the APIs that create and verify just the
```

Comment:
@[dkostic](https://github.com/dkostic) dkostic [27 days ago](https://github.com/awslabs/aws-lc/pull/637/files#r1002064009)
We only want to bring the APIs that create and verify just the signature.

Why is that?

Reply:
We have chosen to select the two APIs `pqcrystals_dilithium3_ref_signature` and `pqcrystals_dilithium3_ref_verify` to be consistent with existing EVP signature functionality. This is also consistent with other implementations, such as within [LibOQS](https://github.com/open-quantum-safe/liboqs/blob/main/src/sig/dilithium/sig_dilithium_3.c#L33-L34).

If we have requirements for the other option (append message) these can be added in later.